web/src/main/java/com/synopsys/integration/alert/web/api/job/JobFieldModelPopulationUtils.java

```diff
@@ -0,0 +1,189 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.web.api.job;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.synopsys.integration.alert.common.enumeration.ConfigContextEnum;
+import com.synopsys.integration.alert.common.persistence.model.job.DistributionJobModel;
+import com.synopsys.integration.alert.common.persistence.model.job.details.AzureBoardsJobDetailsModel;
+import com.synopsys.integration.alert.common.persistence.model.job.details.DistributionJobDetailsModel;
+import com.synopsys.integration.alert.common.persistence.model.job.details.EmailJobDetailsModel;
+import com.synopsys.integration.alert.common.persistence.model.job.details.JiraCloudJobDetailsModel;
+import com.synopsys.integration.alert.common.persistence.model.job.details.JiraServerJobDetailsModel;
+import com.synopsys.integration.alert.common.persistence.model.job.details.MSTeamsJobDetailsModel;
+import com.synopsys.integration.alert.common.persistence.model.job.details.SlackJobDetailsModel;
+import com.synopsys.integration.alert.common.rest.model.FieldModel;
+import com.synopsys.integration.alert.common.rest.model.FieldValueModel;
+import com.synopsys.integration.alert.common.rest.model.JobFieldModel;
+import com.synopsys.integration.alert.descriptor.api.BlackDuckProviderKey;
+
+public class JobFieldModelPopulationUtils {
+    public static JobFieldModel createJobFieldModel(DistributionJobModel jobModel) {
+        FieldModel providerFieldModel = new FieldModel(new BlackDuckProviderKey().getUniversalKey(), ConfigContextEnum.DISTRIBUTION.name(), new HashMap<>());
+        populateProviderFields(providerFieldModel, jobModel);
+
+        FieldModel channelFieldModel = new FieldModel(jobModel.getChannelDescriptorName(), ConfigContextEnum.DISTRIBUTION.name(), new HashMap<>());
+        populateChannelFields(channelFieldModel, jobModel);
+
+        String jobIdString = Optional.ofNullable(jobModel.getJobId())
+                                 .map(UUID::toString)
+                                 .orElse(null);
+        return new JobFieldModel(jobIdString, Set.of(providerFieldModel, channelFieldModel));
+    }
+
+    public static void populateProviderFields(FieldModel providerFieldModel, DistributionJobModel jobModel) {
+        String providerCommonConfigId = Optional.ofNullable(jobModel.getBlackDuckGlobalConfigId())
+                                            .map(String::valueOf)
+                                            .orElse(null);
+        putField(providerFieldModel, "provider.common.config.id", providerCommonConfigId);
```

Comment:
Are all of these constants no longer available in this package? Using the same constants would cut down on any potential misspellings.

Reply:
Ugh lol. I stole a lot of this from the DB util which didn't have access to these field constants. I'll fix it.